milestones/services.py

```diff
@@ -0,0 +1,46 @@
+"""
+A wrapper class around requested methods exposed in api.py
+"""
+
+import types
+
+from milestones import api as milestones_api
+
+
+class MilestonesService(object):  # pylint: disable=too-few-public-methods
+    """
+    An xBlock service for xBlocks to talk to the Milestones subsystem.*
+
+    NOTE: This is a Singleton class. We should only have one instance of it!
+    """
+
+    _instance = None
+
+    REQUESTED_FUNCTIONS = [
+        'get_course_content_milestones',
+    ]
+
+    def __new__(cls, *args, **kwargs):
+        """
+        This is the class factory to make sure this is a Singleton
+        """
+        if not cls._instance:
+            cls._instance = super(MilestonesService, cls).__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self):
+        """
+        Class initializer, which just inspects the libraries and exposes the same functions
+        listed in REQUESTED_FUNCTIONS
+        """
+        self._bind_to_module_functions()
+
+    def _bind_to_module_functions(self):
+        """
+        bind module functions. Since we use underscores to mean private methods, let's exclude those.
+        """
+        for attr_name in self.REQUESTED_FUNCTIONS:
+            attr = getattr(milestones_api, attr_name, None)
```

Comment:
Since I took out the module parameter, I'm only binding the methods from` milestones/api.py`. Since `milestones_helpers.py` and `api.py` don't have exactly the same methods, it might cause some confusion of why `MilestonesService() `doesn't have some of the functions that `milestones_helpers.py` has. Was that expected or how should I go about this?